assembly/src/assembler/instruction/crypto_ops.rs

```diff
@@ -215,17 +160,75 @@ fn read_mtree_node(span: &mut SpanBuilder) {
 /// Update a node in the merkle tree. The `copy` flag will be passed as argument of the `MrUpdate`
 /// operation.
 ///
-/// This operation takes 5 VM cycles.
-fn update_mtree(span: &mut SpanBuilder, copy: bool) {
+/// This operation takes 29 VM cycles.
+fn update_mtree(span: &mut SpanBuilder, copy: bool) -> Result<Option<CodeBlock>, AssemblyError> {
+    // stack: [d, i, R_old, V_new, ...]
+    // output: [R_new, R_old, V_new, V_old, ...]
+
+    // Inject the old node value onto the stack for the call to MRUPDATE.
+    // stack: [V_old, d, i, R_old, V_new, ...] (4 cycles)
+    read_mtree_node(span);
+
     #[rustfmt::skip]
-    span.push_ops([
+    let ops = [
+        // Note: The stack is 14 elements deep already. The existing ops manipulate up to depth 16,
+        // so it's only possible to copy 2-elements at the time.
+
+        // COPY V_old
+        // These instructions will push the current copy of V_old down, and create a new word with
+        // the same elements on the top of the stack.
+        //
+        // ========================================================================================
```

Comment:
question: on my laptop these lines are wrapping, which IMO makes it harder to follow the code while reviewing. I used to keep my line length at 100 characters for this reason (above the usual 79/80 chars). Do we have any rules on that?

Reply:
Yep, we also limit to 100 character line length. Is this line longer than that?

Reply:
hummm, you're right, it isn't. I assumed it was more than 100 because of the wrapping. I wonder why it is wrapping on my screen then, the font is mono

![image](https://user-images.githubusercontent.com/310139/214620645-0bba5b79-5ae9-4df3-b6e1-46b30a366db4.png)